### ISSUE TITLE
Projects with a single module won't open in the source viewer

### ISSUE THREAD

**Issue by axman6** [CrystalSplitter/ghcitui#48]
## Description

I have a couple of projects which have a single module under `src/` (https://github.com/data61/foldl-statistics/ is an example). When running ghcitui, it compiles and loads the project, but pressing `M` then `Enter` or `o` doesn't load the module into the source viewer. I've tried a couple of other projects with multiple source modules and they seem to work fine.

### To Reproduce

```shell
$ git clone https://github.com/data61/foldl-statistics/
$ cd foldl-statistics
$ ghcitui
# press M then Enter or o
```

### Expected behaviour

The source should be loaded in the source viewer.

### Screenshots

<img width="992" alt="Screenshot 2024-11-06 at 15 14 31" src="https://github.com/user-attachments/assets/4099eb0d-ab5a-4cad-bf2f-e5b96e656c3c">

### Environment

- OS: macOS 14.7.1
- GHCiTUI Version: 0.3.0.0, installed via `cabal install ghcitui`)

**Comment by simonmichael** [comment]
I confirm the same problem, which I've seen in larger projects too depending how you start GHCI. ghcitui sometimes lists modules but won't display their code. Here's one attempt, with debug output.
```
~/src/foldl-statistics$ ghcitui -vvv --cmd 'stack exec -- ghci -isrc -isrc/Control/Foldl src/Control/Foldl/Statistics.hs'
Starting up GHCiTUI with: `stack exec -- ghci -isrc -isrc/Control/Foldl src/Control/Foldl/Statistics.hs`...
This can take a while...
[ghcid startup:out] GHCi, version 9.8.2: https://www.haskell.org/ghc/  :? for help
[ghcid startup:out] Loaded GHCi configuration from /Users/simon/.ghci
[ghcid startup:out] [1 of 1] Compiling Control.Foldl.Statistics ( src/Control/Foldl/Statistics.hs, interpreted )
[ghcid startup:out] Ok, one module loaded.
[DEBUG]: |startup| GHCi Daemon initted

[DEBUG]: |updateContext| CMD: :show context

[DEBUG]: |updateContext| OUT:
[DEBUG]: 

[DEBUG]: |updateBindings| CMD: :show bindings

[DEBUG]: |updateBindings| OUT:
[DEBUG]: it :: () = ()
[DEBUG]: 

[DEBUG]: updateModuleFileMap|: CMD: :show modules

[DEBUG]: |updateModuleFileMap|: OUT: Control.Foldl.Statistics ( src/Control/Foldl/Statistics.hs, interpreted )
[DEBUG]: 

[DEBUG]: |updateModuleFileMap| modules: [("Control.Foldl.Statistics","src/Control/Foldl/Statistics.hs")]

[DEBUG]: |updateBreakList| CMD: :show breaks

[DEBUG]: |updateBreakList| OUT:
[DEBUG]: No active breakpoints.

[DEBUG]: |history| OUT:
[DEBUG]: Not stopped at a breakpoint

GHCiTUI has shut down; have a nice day :)
```

Related: #30

**Comment by CrystalSplitter** [comment]
I have identified the issue.

When there's only one module, we auto-select that module to load into the source viewer. However, the source viewer isn't refreshing first time it loads, so it shows the default splash screen. Additionally, the source viewer skips refreshing if the [new module is the same as the old module path](https://github.com/CrystalSplitter/ghcitui/blob/main/lib/ghcitui-brick/Ghcitui/Brick/AppState.hs#L154).

So two solutions I'm thinking:

1. Refresh the source on first load, so that it displays the source immediately if available.
2. Always refresh the source on module select.

Either of these seem fine solutions, though I'm leaning towards 1. Perhaps do both.

**Comment by CrystalSplitter** [comment]
This _should_ be fixed in [c06c1fa](https://github.com/CrystalSplitter/ghcitui/commit/c06c1fa56773a3ae0e0f568b76a6ff6922074ce7). Feel free to build from `develop` if you wish to test it yourself. I am unlikely to release a new version until the weekend (or next weekend), so apologise for the delay.
